gradle-plugin/src/main/java/com/squareup/anvil/plugin/AnvilPlugin.kt

```diff
@@ -8,21 +8,58 @@ import com.android.build.gradle.TestExtension
 import com.android.build.gradle.TestedExtension
 import com.android.build.gradle.api.BaseVariant
 import org.gradle.api.GradleException
-import org.gradle.api.Plugin
 import org.gradle.api.Project
 import org.gradle.api.plugins.AppliedPlugin
 import org.gradle.api.plugins.PluginManager
 import org.gradle.api.provider.Provider
 import org.jetbrains.kotlin.gradle.internal.KaptGenerateStubsTask
 import org.jetbrains.kotlin.gradle.plugin.KaptExtension
+import org.jetbrains.kotlin.gradle.plugin.KotlinCompilation
+import org.jetbrains.kotlin.gradle.plugin.KotlinCompilerPluginSupportPlugin
 import org.jetbrains.kotlin.gradle.plugin.KotlinPluginWrapper
 import org.jetbrains.kotlin.gradle.plugin.PLUGIN_CLASSPATH_CONFIGURATION_NAME
+import org.jetbrains.kotlin.gradle.plugin.SubpluginArtifact
+import org.jetbrains.kotlin.gradle.plugin.SubpluginOption
 import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
+import java.io.File
 import java.util.Locale.US
 import java.util.concurrent.atomic.AtomicBoolean
 
-open class AnvilPlugin : Plugin<Project> {
-  override fun apply(project: Project) {
+open class AnvilPlugin : KotlinCompilerPluginSupportPlugin {
+
+  override fun isApplicable(kotlinCompilation: KotlinCompilation<*>): Boolean {
+    return kotlinCompilation.target.project.plugins.hasPlugin(AnvilPlugin::class.java)
+  }
+
+  override fun getCompilerPluginId(): String = "com.squareup.anvil.compiler"
+
+  override fun getPluginArtifact(): SubpluginArtifact = SubpluginArtifact(
+      groupId = GROUP,
+      artifactId = "compiler",
+      version = VERSION
+  )
+
+  override fun applyToCompilation(
+    kotlinCompilation: KotlinCompilation<*>
+  ): Provider<List<SubpluginOption>> {
+    val project = kotlinCompilation.target.project
+
+    // Notice that we use the name of the Kotlin compile task as a directory name. Generated code
+    // for this specific compile task will be included in the task output. The output of different
+    // compile tasks shouldn't be mixed.
+    val srcGenDir = File(project.buildDir, "anvil/src-gen-${kotlinCompilation.name}")
```

Comment:
Should be using `File.separator` if you want this to be portable for windows users.

Reply:
That was resolved in another PR.